src/modules/sensors/sensors.cpp

```diff
@@ -1542,6 +1506,39 @@ Sensors::parameter_update_poll(bool forced)
 					(void)sprintf(str, "CAL_MAG%u_ZSCALE", i);
 					failed |= (OK != param_get(param_find(str), &gscale.z_scale));
 
+					(void)sprintf(str, "CAL_MAG%u_ROT", i);
+
+					if (ioctl(fd, MAGIOCGEXTERNAL, 0) <= 0) {
+						/* mag is internal */
+						_mag_rotation[s] = _board_rotation;
+						/* reset param to -1 to indicate external mag */
+						int32_t minus_one = MAG_ROT_VAL_INTERNAL;
+						param_set_no_notification(param_find(str), &minus_one);
+					} else {
+
+						int32_t mag_rot = 0;
+						param_get(param_find(str), &mag_rot);
+
+						/* handling of old setups, will be removed */
+						if (s == 0) {
+							int32_t deprecated_mag_rot = 0;
+							param_get(param_find("SENS_EXT_MAG_ROT"), &deprecated_mag_rot);
+
+							/* if the old param is non-zero, set the new one to the same value */
+							if ((deprecated_mag_rot != 0) && (mag_rot <= 0)) {
+								mag_rot = deprecated_mag_rot;
+								param_set_no_notification(param_find(str), &mag_rot);
```

Comment:
Why exactly should there be no notification? Is it because this param is not needed anywhere else?